<?xml version="1.0"?>
<launch>

    <!-- these are the arguments you can pass this launch file -->

    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="Y" default="0"/>

    <!-- Parameters for the local planner-->
    <arg name="random_spawn_dyn_obj" default="false"/>
    <arg name="random_waypoint_generation" default="false"/>

    <arg name="odometry_topic" default="/base/odom"/>


    <!--enable_spray-->
    <arg name="enable_spray" default="false"/>
    <arg name="initial_waypoint_id" default="0"/>

    <!-- Gui Parameter -->
    <param name="use_gui" value="false"/>

    <arg name="tf_prefix" value="rbsherpa_hl_"/>
    <arg name="robot_name" default="rbsherpa_hl_"/>

  
   <include file="$(find rm3_ackermann_controller)/launch/ackermann_controller.launch">
    <arg name="odom_topic" default="$(arg odometry_topic)"/>
   </include>


    <node name="ibvs_controller_node" pkg="pantheon_planning_and_control" type="IBVS_random_controller_node" clear_params="true" output="screen" 
          args="$(find pantheon_planning_and_control)/parameters/params_ackermann.yaml">
    </node>


    <node name="map_manager" pkg="sherpa_ros" type="map_manager_node"/>


    <node name="task_manager" pkg="sherpa_ros" type="task_manager_spray.py" output="screen">
	    <param name="tour" value="tours/ackermann_tour" />
        <param name="map_points" value="ackermann_map" />
        <param name="odometry_topic" value="$(arg odometry_topic)" />
        <param name="enable_spray" value="$(arg enable_spray)" />
        <param name="threshold" value="0.4"/>
        <param name="initial_waypoint_id" value="$(arg initial_waypoint_id)"/>
    </node>

    <node name="spray_controller" pkg="sherpa_ros" type="spray_controller.py" output="screen">
	    <param name="odometry_topic" value="$(arg odometry_topic)" />
        <param name="target_topic" value="/target" />
    </node>

</launch>
